pytorch_lightning/callbacks/base.py

```diff
@@ -282,24 +282,52 @@ def on_keyboard_interrupt(self, trainer: "pl.Trainer", pl_module: "pl.LightningM
     def on_exception(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", exception: BaseException) -> None:
         """Called when any trainer execution is interrupted by an exception."""
 
+    def state_dict(self) -> Dict[str, Any]:
+        """Called when saving a checkpoint, implement to generate callback's ``state_dict``.
+
+        Returns:
+            A dictionary containing callback state.
+        """
+        return {}
+
+    def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
+        """Called when loading a checkpoint, implement to reload callback state given callback's ``state_dict``.
+
+        Args:
+            state_dict: the callback state returned by ``state_dict``.
+        """
+        pass
+
     def on_save_checkpoint(
         self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", checkpoint: Dict[str, Any]
-    ) -> dict:
-        """Called when saving a model checkpoint, use to persist state.
+    ) -> Optional[dict]:
+        r"""
+        .. deprecated:: v1.6
+            Returning state in this callback hook was deprecated in v1.6 and will be removed in v1.8.
+            Use ``state_dict`` instead to return state. In v1.8 ``on_save_checkpoint`` can only return None.
+
+        Called when saving a checkpoint to give you a chance to store anything else you might want to save.
```

Comment:
```suggestion
        r"""
        Called when saving a checkpoint to give you a chance to store anything else you might want to save.
```

Reply:
agree with @Borda , the description of the hook should come first :)